### ISSUE TITLE
Adding Cascade delete to existing entity not generating migrations for update

### ISSUE THREAD

**Issue by sslgeorge** [typeorm/typeorm#1986]
**Issue type:**

[ ] question
[ ✔] bug report
[ ✔] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[✔ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[ ✔] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
1. Create two entities that are related by  a ONE-to-MANY and MANY-to-ONE  
2. Generate the migration for the entities
3. Run the migration
4. Add Cascade delete decorator option to the foreignkey field
5. Generate migration

**Expected Behaviour**
Migration should be generated, the foreign key should be dropped and recreated with the cascade delte

**Actual Behaviour**
The CLI says there are no migrations to create
<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by pleerock** [comment]
> Add Cascade delete decorator

you mean `onDelete: "CASCADE"`, right? Are you using `0.2.0` ?

**Comment by kerenfi** [comment]
@pleerock  any updates?

**Comment by ls-julien-deruere** [comment]
any, I have the same problem with version 0.2.24

**Comment by brunocamboim** [comment]
Same problem here, with version 0.2.31, but using postgres and trying to use onUpdate: cascade

**Comment by SmallhillCZ** [comment]
For me it didn't work when foreign key was already present, but with a "normal" name (e.g. created using Adminer), instead of the hash name.

When I regenerate the whole schema so that foreign keys will have hash names generated by TypeOrm (e.g. FK_879f45e078fdb7435ba0852e99f), then it works.

**Comment by jcapogna** [comment]
I have the same problem.  

It seems like cascades only work if you have them defined when creating a foreign key. Adding them afterward does not update via migrations.

**Comment by RaresM7373** [comment]
Any updatest on this ? Currently facing the same issue with version 0.3.10 ?

**Comment by artursudnik** [comment]
Hey, I've just discovered this issue is still not fixed.

**Comment by franciuma** [comment]
I have the same problem.

To solve it, I dropped my database and recreated it by CREATE DATABASE command.
Then, I used php bin/console doctrine:schema:create. Finally php bin/console doctrine:schema:update.
It worked!

I'm using php with symfony and mysql.

**Comment by MJTheOne** [comment]
> I have the same problem.
> 
> To solve it, I dropped my database and recreated it by CREATE DATABASE command. Then, I used php bin/console doctrine:schema:create. Finally php bin/console doctrine:schema:update. It worked!
> 
> I'm using php with symfony and mysql.

That's really bad advice. What about existing structures (what this issue is about). We can't just drop a production database.

For now I'll just create a migration by hand, but it's a shame this has been an issue for 5 years...

**Comment by franciuma** [comment]
> > I have the same problem.
> > To solve it, I dropped my database and recreated it by CREATE DATABASE command. Then, I used php bin/console doctrine:schema:create. Finally php bin/console doctrine:schema:update. It worked!
> > I'm using php with symfony and mysql.
> 
> That's really bad advice. What about existing structures (what this issue is about). We can't just drop a production database.
> 
> For now I'll just create a migration by hand, but it's a shame this has been an issue for 5 years...

That's true, in my case I have a little project in develop and it doesn't mind if I have to drop my database. Obviusly, this is really dangerous if you have a production environment.
Apologies for this comment, already deleted, thanks for your opinion.

**Comment by bhaugeea** [comment]
You can work around this by using the `onDelete: 'CASCADE'` syntax rather than `cascade: [ 'remove' ]`. Migrations are correctly generated if you do it that way. Still a bug though.

**Comment by versa-dev** [comment]
I am using `0.3.17`, getting the same issue.
I think this is a kind of critical issue of this module, and not sure why this isn't fixed yet though it has been long time.

**Comment by LucasMoskun** [comment]
On version `0.3.11` with postrgres I am experiencing a similar issue. When removing `onDelete: 'CASCADE` from an entity, generating does not recognize any changes

**Comment by wppaing** [comment]
I am facing the same issue on `0.3.20`.

**Comment by artursudnik** [comment]
This is going to get 6 years old soon! :)

**Comment by kisstamasj** [comment]
i experienced the same issue

**Comment by alihut** [comment]
Can't believe this issue is not resolved yet. I'm having the same problem.

**Comment by matiasgarcia** [comment]
This is open source software, you can help to implement the features rather than complaining they are not solved.

**Comment by anipnwr7777** [comment]
A lot of people seem to be running into this issue. I've added tests to verify that what @AntoniusGolly mentioned is correct: we need to add onDelete: 'CASCADE' on the owning side of the relationship for migrations to be generated properly.

I don't believe TypeORM intends to generate database-level cascade constraints when only cascade: true is specified. If that assumption is correct, I think the documentation should be updated, as this appears to be a common source of confusion. - @pkuczynski need your input on this and I can update https://github.com/typeorm/typeorm/pull/12553 accordingly and we can close this issue.
